sdk/search/azure-search/src/main/java/com/azure/search/models/FieldMapping.java

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+
+package com.azure.search.models;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.search.implementation.models.FieldMappingFunction;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * Defines a mapping between a field in a data source and a target field in an
+ * index.
+ */
+@Fluent
+public final class FieldMapping {
+    /*
+     * The name of the field in the data source.
+     */
+    @JsonProperty(value = "sourceFieldName", required = true)
+    private String sourceFieldName;
+
+    /*
+     * The name of the target field in the index. Same as the source field name
+     * by default.
+     */
+    @JsonProperty(value = "targetFieldName")
+    private String targetFieldName;
+
+    /*
+     * A function to apply to each source field value before indexing.
+     */
+    @JsonProperty(value = "mappingFunction")
+    private FieldMappingFunction mappingFunction;
+
+    /**
+     * Get the sourceFieldName property: The name of the field in the data
+     * source.
+     *
+     * @return the sourceFieldName value.
+     */
+    public String getSourceFieldName() {
+        return this.sourceFieldName;
+    }
+
+    /**
+     * Set the sourceFieldName property: The name of the field in the data
+     * source.
+     *
+     * @param sourceFieldName the sourceFieldName value to set.
+     * @return the FieldMapping object itself.
+     */
+    public FieldMapping setSourceFieldName(String sourceFieldName) {
+        this.sourceFieldName = sourceFieldName;
+        return this;
+    }
+
+    /**
+     * Get the targetFieldName property: The name of the target field in the
+     * index. Same as the source field name by default.
+     *
+     * @return the targetFieldName value.
+     */
+    public String getTargetFieldName() {
+        return this.targetFieldName;
+    }
+
+    /**
+     * Set the targetFieldName property: The name of the target field in the
+     * index. Same as the source field name by default.
+     *
+     * @param targetFieldName the targetFieldName value to set.
+     * @return the FieldMapping object itself.
+     */
+    public FieldMapping setTargetFieldName(String targetFieldName) {
+        this.targetFieldName = targetFieldName;
+        return this;
+    }
+
+    /**
+     * Get the mappingFunction property: A function to apply to each source
+     * field value before indexing.
+     *
+     * @return the mappingFunction value.
+     */
+    public FieldMappingFunction getMappingFunction() {
```

Comment:
This has public API which returns an implementation class.

Reply:
FieldMappingFunction moved back under models (removed from implementation) - autorest config will be updated as well as this is a generated class

Reply:
@mssfang Could we add some unit tests to the Checkstyle rule which should be catching implementation in public API to validated these cases will be found.

Reply:
Yes. Create an issue for it : https://github.com/Azure/azure-sdk-for-java/issues/7080